Issue 1824 Girder deconstruction doesn't make tile passable on serverside

JaytleBee opened the pull request:
### Purpose
Fixes #1824 

### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- [X]  I correctly commented my code (Not much to comment)
- [X]  My code is indented with tabs and not spaces
- [X]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [X]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### Notes:
Turns out to be a pretty simple fix: The girder doesn't actually get destroyed (because pooling) so we need to make it passable when it disappears.

I also fixed a typo in a related file

JaytleBee committed:
1824 Girder deconstruction doesn't make tile passable on serverside

Aranclanos reviewed (changes requested):
Hacky fix, the girder should not be in the tile at all, will bring issues later on. Fix it properly.

JaytleBee committed:
Fixed #1824 in a more proper way

It seems that the server used the client's predicetd position to determine
whether a tile is active or not. Now it uses the server's position.
I assume someone copypasted the function and forgot to change a line.

fomalsd commented on UnityProject/Assets/Scripts/Items/MetalTrigger.cs:
```diff
@@ -9,7 +9,7 @@
 {
 	private bool isBuilding;
-	public GameObject griderPrefab;
+	public GameObject girderPrefab;
```
are you sure prefab didn't lose reference because of that change?
use [FormerlySerializedAs("griderPrefab")] above it to keep the references

JaytleBee replied:
Well, MetalTrigger is only used for the metal prefab and I updated the reference there, so should I still add the FormerlySerializedAs (and should I remove the .prefab from the commit)

Aranclanos replied:
just grab a piece of metal ingame and see if it works

JaytleBee replied:
It does, I tested my commit

fomalsd reviewed (changes requested):
Yeah, it was a typo. But please add FormerlySerializedAs attribute to girderPrefab to keep the existing references after rename
